cdap-api/src/main/java/co/cask/cdap/api/mapreduce/MapReduceContext.java

```diff
@@ -63,7 +64,9 @@
    * Overrides the input configuration of this MapReduce job to use the specific stream.
    *
    * @param stream the input stream.
+   * @deprecated as of 3.4.0. Use {@link MapReduceContext#addInput(Input)} instead.
    */
+  @Deprecated
   void setInput(StreamBatchReadable stream);
```

Comment:
From the implementation, it seems like exception will be raised from the `addInput` call after any of these `setInput` was called. This behavior should be documented as well.